controller-server/src/test/java/com/yahoo/vespa/hosted/controller/ZoneRegistryMock.java

```diff
@@ -81,18 +82,33 @@ public boolean hasZone(ZoneId zoneId) {
 
     @Override
     public Optional<URI> getLogServerUri(DeploymentId deploymentId) {
-        return Optional.of(deploymentId.zoneId())
-                .map(z -> URI.create(String.format("http://log.%s.test", deploymentId.zoneId().value())));
+        if ( ! hasZone(deploymentId.zoneId()))
+            return Optional.empty();
+
+        String kibanaQuery = "/#/discover?_g=()&_a=(columns:!(_source)," +
+                             "index:'logstash-*',interval:auto," +
+                             "query:(query_string:(analyze_wildcard:!t,query:'" +
+                             "HV-tenant:%22" + deploymentId.applicationId().tenant().value() + "%22%20" +
+                             "AND%20HV-application:%22" + deploymentId.applicationId().application().value() + "%22%20" +
+                             "AND%20HV-region:%22" + deploymentId.zoneId().region().value() + "%22%20" +
+                             "AND%20HV-instance:%22" + deploymentId.applicationId().instance().value() + "%22%20" +
+                             "AND%20HV-environment:%22" + deploymentId.zoneId().environment().value() + "%22'))," +
+                             "sort:!('@timestamp',desc))";
+
+        URI kibanaPath = URI.create(kibanaQuery);
+        return Optional.of(URI.create(String.format("http://log.%s.test", deploymentId.zoneId().value())).resolve(kibanaPath));
     }
 
     @Override
     public Duration getDeploymentTimeToLive(ZoneId zoneId) {
-        return deploymentTimeToLive.get(zoneId);
+        return deploymentTimeToLive.containsKey(zoneId)
+                ? deploymentTimeToLive.get(zoneId)
+                : Duration.ofMillis(Long.MAX_VALUE / 2);
```

Comment:
The default TTL value is hard to understand. I would not interpret it as infinite, but as a specific default value. `Optional<Duration>` seems more obvious.

Reply:
Agree. Why change the signature? Most deployments do not have a TTL, so optional makes more sense.